### ISSUE TITLE
PS7: Get-Item and Get-ChildItem slow on UNC paths because it enumerates parent directories

### ISSUE THREAD

**Issue by jgard** [PowerShell/PowerShell#12795]
Enterprise environment, have a central file server with home drives for each employee (~4,000).  Each employee has a path like `\\SERVER.DOMAIN.COM\share\home\EMPLOYEEID`, and also has it mapped to `H:\`.

In Powershell 7.01, cmdlets like `Get-Item` and `Get-ChildItem` take far longer to run against the UNC path than the corresponding mapped drive path.  Powershell 6.x and 5.1 do not exhibit this.

Tools like Wireshark and Process Monitor suggest that pwsh is traversing the path, getting a full directory list at each level.  For example, if I run `Get-Item \\SERVER.DOMAIN.COM\share\home\MYID\folder`, I will see in Wireshark all employee ID folder names in packet data as it enumerates at the `\\SERVER.DOMAIN.COM\share\home` level.

# Steps to reproduce
```powershell
Get-Item \\SERVER.DOMAIN.COM\share\home\MYID\folder  #Slow
Get-Item H:\folder  #Fast
```

# Expected behavior

```powershell
#Only perform the necessary SMB request.  Do not enumerate parent folders
Get-Item \\SERVER.DOMAIN.COM\share\home\MYID\folder
```

# Environment data
```none
Name                           Value
----                           -----
PSVersion                      7.0.1
PSEdition                      Core
GitCommitId                    7.0.1
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

Also have PS6 and PS5 installed.
```

**Comment by miketheitguy** [comment]
Could this be related to the Windows setting with path traversal?

https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/bypass-traverse-checking

**Comment by miketheitguy** [comment]
It also looks like there's some work on the FileSystemProvider that might show up here in a future build? https://github.com/PowerShell/PowerShell/blob/9212aac0fa020b657454ef4f9ff2fdd6fc6d759c/src/System.Management.Automation/engine/ExperimentalFeature/ExperimentalFeature.cs

It doesn't appear that PSFileSystemProviderV2 is in 7.1-preview3 just yet. So maybe that will help :P

**Comment by jgard** [comment]
> Could this be related to the Windows setting with path traversal?

@miketheitguy Not sure how a User Rights Assignment security setting relating to local NTFS could relate to behavior between different versions of Powershell and their interaction with a remote file share (which happens to be on a NetApp), but I'm open to any suggestions!

Are you in a position to reproduce my symptom?

**Comment by mattcargile** [comment]
I see this behavior with _NetApp_ as well. It is pretty painful. Tab completion is mostly unusable and takes quite some time on big directories like with a Citrix XenApp Profiles share. I wonder if it is some setting with NetApp. `C$` shares on Windows Servers aren't as bad.

**Comment by jgard** [comment]
Hmm, I'm unable to reproduce this now on 7.3.4.  I suppose it can be closed.

**Comment by cbucher** [comment]
I can reproduce with PowerShell 7.4.6.

Using Procmon, I locate the problem in usage of QueryDirectory operation (with FileInformationClass: FileFullDirectoryInformation) on share directory (this call takes a long time when there are many subdirectories).

When if use the command `Get-ChildItem \\SERVER.DOMAIN.COM\share\a\b`, it spend a lot of time in QueryDirectory of `\\SERVER.DOMAIN.COM\share`.

With PowerShell 5.1, there is no such behavior.

**Comment by endo64** [comment]
This also happens to me with a Network mapped drive:

`X:\a\b` is mapped to a network share.
`X:\a` has many files in it, yet `X:\a\b` has only a few files. 

```powershell
cd X:\a\b  # very slow
ls .\  # very slow even though there are 2-3 files in current directory, but there are many in parent dir.
```

**Comment by colejohnson66** [comment]
Path traversal is still a problem on 7.6.1.

**Comment by aurnor** [comment]
This has been fixed on 7.6.3, which relies on .NET 10.0.9 where the issue was fixed.
[What's New in PowerShell 7.6 - PowerShell | Microsoft Learn](https://learn.microsoft.com/en-us/powershell/scripting/whats-new/what-s-new-in-powershell-76?view=powershell-7.6): _PowerShell 7.6.3 is built on the .NET 10.0.9 runtime._
